src/features/zaaksysteem/service.ts

```diff
@@ -43,32 +44,46 @@ export const useZakenByZaaknummer = (zaaknummer: Ref<string>) => {
   return ServiceResult.fromFetcher(getUrl, overviewFetcher);
 };
 
+const singleZaakFetcher = function fetcher(url: string): Promise<ZaakDetails> {
+  return fetchLoggedIn(url)
+    .then(throwIfNotOk)
+    .then((x) => x.json())
+    .then(mapZaakDetails);
+};
+
 export const useZaakById = (id: Ref<string>) => {
   const getUrl = () => getZaakUrl(id.value);
-
-  function fetcher(url: string): Promise<ZaakDetails> {
-    return fetchLoggedIn(url)
-      .then(throwIfNotOk)
-      .then((x) => x.json())
-      .then(mapZaakDetails);
-  }
-
-  return ServiceResult.fromFetcher(getUrl, fetcher);
+  return ServiceResult.fromFetcher(getUrl, singleZaakFetcher);
 };
 
 export const useZakenByVestigingsnummer = (vestigingsnummer: Ref<string>) => {
   const getUrl = () => {
     if (!vestigingsnummer.value) return "";
     const url = new URL(location.href);
-    url.pathname = zaaksysteemBaseUri + "/zaken";
+    url.pathname = zaaksysteemBaseUri + "/rollen";
     url.searchParams.set(
-      "embedded.rollen.embedded.betrokkeneIdentificatie.vestigingsNummer",
+      "betrokkeneIdentificatie__vestiging__vestigingsNummer",
       vestigingsnummer.value
     );
     return url.toString();
   };
 
-  return ServiceResult.fromFetcher(getUrl, overviewFetcher);
+  const fetcher = (url: string) =>
+    fetchLoggedIn(url)
+      .then(throwIfNotOk)
+      .then(parseJson)
+      .then((r) =>
+        parsePagination(r, async (x) => {
+          const split = (x as RolType).zaak.split("/");
+          const id = split[split.length - 1];
+          const url = getZaakUrl(id);
+          const result = await singleZaakFetcher(url);
+          mutate(url, result);
```

Comment:
wat mutaten we hier?

Reply:
Het cache record voor de zaak, met als key de url van de zaak. Als je doorklikt naar een zaak, zie je daardoor niet eerst een spinner.

Reply:
onduidelijke naam, maar die komt blijkbaar rechtstreeks uit de lib